base/sysinfo.jl

```diff
@@ -532,7 +533,26 @@ including e.g. a WebAssembly JavaScript embedding in a web browser.
 """
 isjsvm(os::Symbol) = (os === :Emscripten)
 
-for f in (:isunix, :islinux, :isbsd, :isapple, :iswindows, :isfreebsd, :isopenbsd, :isnetbsd, :isdragonfly, :isjsvm)
+"""
+    Sys.detectwsl([os])
+
+Runtime predicate for testing if Julia is running inside
+Windows Subsystem for Linux (WSL).
+
+!!! note
+    Unlike `Sys.iswindows`, `Sys.islinux` etc., this is a runtime tests, and thus
+    cannot meaningfully be used in `@static if ` constructs.
+
+!!! compat "Julia 1.12"
+    This function requires at least Julia 1.12.
+"""
+function detectwsl(os::Symbol)
+    islinux(os) &&
+    isfile("/proc/sys/kernel/osrelease") &&
+    contains(read("/proc/sys/kernel/osrelease", String), r"Microsoft|WSL"i)
+end
+
+for f in (:detectwsl, :isunix, :islinux, :isbsd, :isapple, :iswindows, :isfreebsd, :isopenbsd, :isnetbsd, :isdragonfly, :isjsvm)
```

Comment:
```suggestion
for f in (:isunix, :islinux, :isbsd, :isapple, :iswindows, :isfreebsd, :isopenbsd, :isnetbsd, :isdragonfly, :isjsvm)
```
This should fix the error during building. As you noted, `isfile` is used elsewhere in this file, but the difference is that here we're attempting to actually call `detectwsl(KERNEL)` inside of this loop but can't because this file is included in the build before the one that defines `isfile`. (It's also not correct to have it in the loop here anyway since, as you note in the docstring, it's a runtime check that can't be statically evaluated.)